src/components/input/input.jsx

```diff
@@ -1,37 +1,65 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import cautionSvg from '../../svgs/icon-caution-fill.svg';
+import Icon from '../icon/icon.jsx';
 import styles from './input.css';
-import labelPropType from './label-prop-type';
 
-/**
- * Base input component. Accepts any prop that an input can take.
- * @see [RFC](https://github.com/mavenlink/rfc/pull/75)
- * @see [Design spec](https://zpl.io/295Yp7p)
- */
-export default function Input({ 'aria-labelledby': ariaLabelledBy, className, id, ...rest }) {
+function getLabelClassName(className, invalid) {
+  if (className) return className;
+  return invalid ? styles['invalid-label'] : styles.label;
+}
+
+function getClassName(className, invalid) {
+  if (className) return className;
+  return invalid ? styles['invalid-input'] : styles.input;
+}
+
+export default function Input(props) {
   return (
-    <input
-      {...rest}
-      aria-labelledby={ariaLabelledBy}
-      className={className}
-      id={id}
-    />
+    <React.Fragment>
+      <label htmlFor={props.id} className={getLabelClassName(props.cssLabel, props.invalid)}>
```

Comment:
How are ID collisions being handled? It looks like this API places the burden entirely on the consumer with no recommendation. Realistically, are they going to use a uuid generator and prepend something like `<GENERATED-UUID>-meaingful-id-name`? I think should think about this and address it to make this as frictionless as possible for MDS consumers.

Reply:
At the moment, I have a use-case to have a consumer-provided ID API. Kinda weird, huh? But that is the current real world. 

I think it's easy for others to provide their own unique ID that is meaningful to them. For example, if you are building a feature in Mavenlink, you should probably prefix your IDs with your feature name. 

It might make sense to break this API in the future though. I think then it would be ideal to provide a "migration" guide or something.

Reply:
Yeah, as I started on the file picker and thought about this more, I agree that while it's burdensome for consumer (it's burdensome for whoever though), it really should be an injected dependency. On the component level it could be argued the component should generate it's own uuid or similar, but I'm fine with the container or consumer doing so. Bigger stuff can leverage server data ids of course, but at the component level I think `feature-generateduuid` or similar is how folks will have to go perhaps.